utils/hash.go

```diff
@@ -0,0 +1,22 @@
+package utils
+
+import (
+	"crypto/md5"
+	"encoding/hex"
+	"encoding/json"
+)
+
+// GenerateJSONMD5Hash marshals the object into JSON and generates its md5 hash
+func GenerateJSONMD5Hash(obj any) (string, error) {
+	data, err := json.Marshal(obj)
+	if err != nil {
+		return "", err
+	}
+
+	hash := md5.Sum(data)
+	if err != nil {
+		return "", err
+	}
+
+	return hex.EncodeToString(hash[:]), nil
+}
```

Comment:
Please this to commons/hash, this snippet is used/repeated in other repos too

Reply:
cool. will try to move as much utils from other repos as well to commons.

Reply:
https://github.com/flanksource/commons/pull/72